sycl/doc/extensions/experimental/sycl_ext_oneapi_tangle.asciidoc

```diff
@@ -0,0 +1,377 @@
+the same control flow.
+
+
+==== Creation
+
+A new `tangle` can only be created by partitioning an existing group, using the
+`entangle` free-function.
+
+[source, c++]
+----
+namespace ext::oneapi::experimental {
+
+template <typename ParentGroup>
+tangle<ParentGroup> entangle(ParentGroup group);
+
+} // namespace ext::oneapi::experimental
+----
+
+_Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>> &&
+std::is_same_v<Group, sycl::sub_group>` is true.
+
+_Effects:_ Blocks until all work-items in `parent` that will reach this
+synchronization point have reached this synchronization point.
+
+_Synchronization:_ The call in each work-item happens before any work-item
+blocking on the same synchronization point is unblocked.
+Synchronization operations used by an implementation must respect the memory
+scope reported by `ParentGroup::fence_scope`.
+
+_Returns_: A `tangle` consisting of all work-items in `parent` which will reach
+this synchronization point in the same control flow.
+
+NOTE: This function provides stronger guarantees than
+`get_opportunistic_group()`, which returns a group consisting of _some_ of the
+work-items in `parent` which will reach the synchronization point.
+
+
+==== `tangle` Class
+
+[source, c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename ParentGroup>
+class tangle {
+public:
+  using id_type = id<1>;
+  using range_type = range<1>;
+  using linear_id_type = uint32_t;
+  static constexpr int dimensions = 1;
+  static constexpr sycl::memory_scope fence_scope = ParentGroup::fence_scope;
+
+  id_type get_group_id() const;
+
+  id_type get_local_id() const;
+
+  range_type get_group_range() const;
+
+  range_type get_local_range() const;
+
+  linear_id_type get_group_linear_id() const;
+
+  linear_id_type get_local_linear_id() const;
+
+  linear_id_type get_group_linear_range() const;
+
+  linear_id_type get_local_linear_range() const;
+
+  bool leader() const;
+};
+
+}
+----
+
+[source,c++]
+----
+id_type get_group_id() const;
+----
+_Returns_: An `id` representing the index of the tangle.
+
+NOTE: This will always be an `id` with all values set to 0, since there can
+only be one tangle.
+
+[source,c++]
+----
+id_type get_local_id() const;
+----
+_Returns_: An `id` representing the calling work-item's position within
+the tangle.
+
+[source,c++]
+----
+range_type get_group_range() const;
+----
+_Returns_: A `range` representing the number of tangles.
+
+NOTE: This will always return a `range` of 1 as there can only be one
+tangle.
+
+[source,c++]
+----
+range_type get_local_range() const;
+----
+_Returns_: A `range` representing the number of work-items in the tangle.
+
+[source,c++]
+----
+id_type get_group_linear_id() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_group_id()`.
+
+[source,c++]
+----
+id_type get_local_linear_id() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_local_linear_id()`.
```

Comment:
```suggestion
_Returns_: A linearized version of the `id` returned by `get_local_id()`.
```

Reply:
I think I fixed all of these at the same time as https://github.com/intel/llvm/commit/cc24795326e7da667c927e701ec472e93ec67d6b, too, so I can't commit the suggestions.  Could you please double-check that I got them all, and mark these comments as resolved if I did?